src/builder/commands/copy.rs

```diff
@@ -159,11 +185,33 @@ impl BuildStep for Copy {
     }
 }
 
-fn hash_path(hash: &mut Digest, path: &Path,
-    filter: &Filter, owner_uid: Option<uid_t>, owner_gid: Option<gid_t>)
+fn reset_permissions(path: &Path, umask: u32)
+    -> Result<(), StepError>
+{
+    let stat = try!(path.symlink_metadata()
```

Comment:
1. In this case I should patch `file_util::copy` & `file_util::shallow_copy` functions. Possible is it worth to add `mode` optional parameter to these functions?
2. Only `umask` from `Copy` should be applied. We only take executable bit from created file.